### PR TITLE
[Feature](sink) support parallel result sink

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
## Proposed changes

support parallel result sink

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by BiteTheDDDDt** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/qe/Coordinator.java [hunk 1]
```diff
@@ -1505,8 +1520,10 @@ private void cancelLatch() {
     }
 
     private void cancelInternal(Types.PPlanFragmentCancelReason cancelReason) {
-        if (null != receiver) {
-            receiver.cancel(cancelReason);
+        if (receivers.isEmpty()) {
```
Comment (Gabriel39) [inline fe/fe-core/src/main/java/org/apache/doris/qe/Coordinator.java:1523]:
```suggestion
        if (!receivers.isEmpty()) {
```

File: fe/fe-core/src/main/java/org/apache/doris/qe/Coordinator.java [hunk 2]
```diff
@@ -1325,13 +1333,13 @@ private void updateStatus(Status status) {
...
     public RowBatch getNext() throws Exception {
-        if (receiver == null) {
+        if (receivers.isEmpty()) {
             throw new UserException("There is no receiver.");
         }
 
         RowBatch resultBatch;
         Status status = new Status();
-        resultBatch = receiver.getNext(status);
+        resultBatch = receivers.get(receivers.size() - 1).getNext(status);
```
Comment (Gabriel39) [inline fe/fe-core/src/main/java/org/apache/doris/qe/Coordinator.java:1342]: Maybe round-robin ?

File: fe/fe-core/src/main/java/org/apache/doris/qe/ResultReceiver.java [hunk 3]
```diff
@@ -84,7 +84,7 @@ public RowBatch getNext(Status status) throws TException {
         try {
             while (!isDone && !isCancel()) {
                 InternalService.PFetchDataRequest request = InternalService.PFetchDataRequest.newBuilder()
-                        .setFinstId(finstId)
```
Comment (Gabriel39) [inline fe/fe-core/src/main/java/org/apache/doris/qe/ResultReceiver.java:87]: Keep compatibility

File: fe/fe-core/src/main/java/org/apache/doris/nereids/properties/RequestPropertyDeriver.java [hunk 4]
```diff
@@ -142,7 +142,7 @@ public Void visitPhysicalHiveTableSink(PhysicalHiveTableSink<? extends Plan> hiv
 
     @Override
     public Void visitPhysicalResultSink(PhysicalResultSink<? extends Plan> physicalResultSink, PlanContext context) {
-        addRequestPropertyToChildren(PhysicalProperties.GATHER);
+        addRequestPropertyToChildren(PhysicalProperties.ANY);
```
Comment (Gabriel39) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/properties/RequestPropertyDeriver.java:145]: If `order by` exists in this query ,we need a `GATHER` exchange opeartor to do global merge sort here.

**Review by Gabriel39 — APPROVED** [review]

**Review by Gabriel39 — CHANGES_REQUESTED** [review]
request changes

**Pull request closed without merging**